CHANGELOG.md

```diff
@@ -1,3 +1,10 @@
+## 1.2.0
+  - Fix Trailing Delimiters requires a false field. A skip field is
+  automatically added when a final delimiter is detected in the dissect pattern.
+  This requires that strict delimiter finding is enforced  - meaning a "no match"
+  results if every delimiter is not found in exactly the declared order
```

Comment:
One can argue that it is a bug fix to an expectation mismatch.
We say in the docs that Dissect scans for a set of delimiters from left to right - the user is made to understand that this is strict but we weren't.

We should discuss this.

If it is considered a breaking change then we can't release it until 7.0.0 because the plugin will have be at 2.0.0. This seems quite artificial to me because after 7.0.0 is released a 6.X user can update and not have the opportunity to see a breaking changes warning.

Reply:
> If it is considered a breaking change then we can't release it until 7.0.0 because the plugin will have be at 2.0.0

**If** the breaking change is necessary we can release it now but we should fork the current major into a branch so we can continue to release bugfixes/features in 1.x

> This seems quite artificial to me because after 7.0.0 is released a 6.X user can update and not have the opportunity to see a breaking changes warning.

The update process will show that a plugin was upgraded to a new major, at that point something will likely break. Ideally the plugin manager would display any breaking change notice, but that's not possible right now.

Reply:
One question, what is our definition of a breaking change?
1. This change will break every installation.
2. This change might break a some installations (identifiable).
3. This change might break a very small number of installations (unidentifiable).

Reply:
@jsvd and I decided that this is not a breaking change because the docs are explicit about having delimiters in between fields and it is not specific about the behaviour when a delimiter is not found in a particular place.

Reply:
👍